File: go/examples/hello_web.rb
require 'webrick'

server = WEBrick::HTTPServer.new :Port => 8080

server.mount_proc '/' do |req, res|
  name = req.path == '/' ? "World" : req.path[1..-1]
  res.body = "Hello, #{name}!"
end

trap 'INT' do server.shutdown end
server.start

File: go/examples/hello_web.go
package main

import "fmt"
import "net/http"

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
		name := req.URL.Path[1:]
		if name == "" {
			name = "World"
		}
		w.Write([]byte(fmt.Sprintf("Hello, %s!", name)))
	})

	fmt.Println("Running on :8080")
	http.ListenAndServe(":8080", nil)
}
